ios/src/ImageCropPicker.m

```diff
@@ -908,4 +723,403 @@ - (void)cropViewController:(TOCropViewController *)cropViewController didFinishC
+                            });
+                        }];
+                    }];
+                }
+            }
+        }];
+    } else {
+        PHAsset *phAsset = assets[0];
+
+        [self showActivityIndicator:^(UIActivityIndicatorView *indicatorView, UIView *overlayView) {
+            if (phAsset.mediaType == PHAssetMediaTypeVideo) {
+                [self getVideoAsset:phAsset completion:^(NSDictionary* video) {
+                    dispatch_async(dispatch_get_main_queue(), ^{
+                        [indicatorView stopAnimating];
+                        [overlayView removeFromSuperview];
+                        [imagePickerController dismissViewControllerAnimated:YES completion:[self waitAnimationEnd:^{
+                            if (video != nil) {
+                                self.resolve(video);
+                            } else {
+                                self.reject(ERROR_CANNOT_PROCESS_VIDEO_KEY, ERROR_CANNOT_PROCESS_VIDEO_MSG, nil);
+                            }
+                        }]];
+                    });
+                }];
+            } else {
+                [phAsset requestContentEditingInputWithOptions:nil
+                                             completionHandler:^(PHContentEditingInput * _Nullable contentEditingInput, NSDictionary * _Nonnull info) {
+                    [manager requestImageDataForAsset:phAsset
+                                              options:options
+                                        resultHandler:^(NSData *imageData, NSString *dataUTI, UIImageOrientation orientation, NSDictionary *info) {
+                        NSURL *sourceURL = contentEditingInput.fullSizeImageURL;
+                        NSDictionary* exif;
+                        if([self.options[@"includeExif"] boolValue]) {
+                            exif = [[CIImage imageWithData:imageData] properties];
+                        }
+
+                        dispatch_async(dispatch_get_main_queue(), ^{
+                            [indicatorView stopAnimating];
+                            [overlayView removeFromSuperview];
+
+                            [self processSingleImagePick:[UIImage imageWithData:imageData]
+                                                withExif: exif
+                                      withViewController:imagePickerController
+                                           withSourceURL:[sourceURL absoluteString]
+                                     withLocalIdentifier:phAsset.localIdentifier
+                                            withFilename:[phAsset valueForKey:@"filename"]
+                                        withCreationDate:phAsset.creationDate
+                                    withModificationDate:phAsset.modificationDate];
+                        });
+                    }];
+                }];
+            }
+        }];
+    }
+}
+
+- (ImageResult *)makeResultFromImageData:(NSData *)data image:(UIImage *)image {
```

Comment:
Moved this into a method so that the `PHPickerViewControllerDelegate` implementation can re-use it